airflow/models/dag.py

```diff
@@ -2755,25 +2755,28 @@ def bulk_write_to_db(
             session.add(orm_dag)
             orm_dags.append(orm_dag)
 
-        # Get the latest dag run for each existing dag as a single query (avoid n+1 query)
-        most_recent_subq = (
-            session.query(DagRun.dag_id, func.max(DagRun.execution_date).label("max_execution_date"))
-            .filter(
-                DagRun.dag_id.in_(existing_dags),
-                or_(DagRun.run_type == DagRunType.BACKFILL_JOB, DagRun.run_type == DagRunType.SCHEDULED),
+        most_recent_runs = {}
+        num_active_runs = {}
+        # Save runtime if dag is not scheduleable
+        if any([dag for dag in dags if not isinstance(dag.timetable, NullTimetable)]):
```

Comment:
Actually I think it is not what I asked for, but likely this is even more readable:

```suggestion
        if not all([dag.isinstance(dag.timetable, NullTimetable) for dag in dags]):
```

BTW. Can you also please add a unit test actually testing it?

Reply:
As PR #30706 changes the structure of ǸullTimeTable` and review feedback was pointing to not using `isinstance()` I propose to wait until #30706 is merged and then re-base this change accordingly.

Reply:
As PR https://github.com/apache/airflow/pull/30706 is merged I changed the logic accordingly and added a unit test for it.